AmplifyPlugins/Core/AWSPluginsCore/Model/Support/Model+GraphQL.swift

```diff
@@ -162,24 +162,30 @@ extension Model {
         }
 
         let fieldNames = getFieldNameForAssociatedModels(modelField: field)
+
+        let foreignKeyValues = fieldNames.map {
+            getFieldValue(for: $0, modelSchema: self.schema)?.flatMap({$0 as? Persistable})
+        }
+
         var values = getModelIdentifierValues(from: value, modelSchema: associateModelSchema)
 
         if fieldNames.count != values.count {
-            // if the field is required, the associated field keys and values should match
-            if field.isRequired {
-                preconditionFailure(
-                """
-                Associated model target names and values for field \(field.name) of model \(modelName) mismatch.
-                There is a possibility that is an issue with the generated models.
-                """
-                )
-            } else if mutationType == .update {
-                // otherwise, pad the values with `nil` to account for removals of associations on updates.
-                values = [Persistable?](repeating: nil, count: fieldNames.count)
+            let defaultValues = [Persistable?](repeating: nil, count: fieldNames.count)
```

Comment:
Do we still need the check on the `mutationType` that was removed in this PR? 

Some context on this: create mutations don't have to send `defaultValues`, only on update mutations so that it acts as a removal of the association. No-op if there is no association originally. Let's take Comment, Post with CPK, example

if it is a create mutation, the comment object translated to the graphQL input will fail when `null` values are sent for the post foreign key fields
```
// this will fail on AppSync, trying to set foreign key to null
{
   "id": "commentId123"
   "postId": null,
   "postTitle": null
}
```

For create mutations, we don't pad the values with null so that only the fields that are being persisted are sent:
```
{
   "id": "commentId123"
}
```
There was also an upper level check done in [here](https://github.com/aws-amplify/amplify-swift/pull/2701/files#r1093600308) so you may want to rebase with latest `data-dev-preview` to get the latest changes, so I wonder if this check is still required.

For update mutations, we do want to pad it with `nil` values:
```
// this will fail on AppSync, trying to set foreign key to null
{
   "id": "commentId123"
   "postId": null,
   "postTitle": null
}
```
When the comment is not associated with a post, then `null` updates are okay since no change in data. When the comment is deassociated with a post, then the `null` values will update the existing values in DynamoDB to `null`